tests/vmlifecycle_test.go

```diff
@@ -442,6 +442,108 @@ var _ = Describe("Vmlifecycle", func() {
+					// If the cfgMap is missing, default to allowEmulation=false
+					// no other error is expected
+					if !errors.IsNotFound(err) {
+						Expect(err).ToNot(HaveOccurred())
+					}
+				}
+				if !allowEmuation {
+					Skip("Software emulation is not enabled on this cluster")
+				}
+			})
+
+			It("should enable emulation in virt-launcher", func() {
+				err := virtClient.RestClient().Post().Resource("virtualmachines").Namespace(tests.NamespaceTestDefault).Body(vm).Do().Error()
+				Expect(err).To(BeNil())
+
+				listOptions := metav1.ListOptions{}
+				var pod k8sv1.Pod
+
+				Eventually(func() error {
+					podList, err := virtClient.CoreV1().Pods(tests.NamespaceTestDefault).List(listOptions)
+					Expect(err).ToNot(HaveOccurred())
+					for _, item := range podList.Items {
+						if strings.HasPrefix(item.Name, vm.ObjectMeta.GenerateName) {
+							pod = item
+							return nil
+						}
+					}
+					return fmt.Errorf("Associated pod for VM '%s' not found", vm.Name)
+				}, 75, 0.5).Should(Succeed())
+
+				emulationFlagFound := false
+				computeContainerFound := false
+				for _, container := range pod.Spec.Containers {
+					if container.Name == "compute" {
+						computeContainerFound = true
+						for _, cmd := range container.Command {
+							By(cmd)
+							if cmd == "--allow-emulation" {
+								emulationFlagFound = true
+							}
+						}
+					}
+				}
+
+				Expect(computeContainerFound).To(BeTrue(), "Compute container was not found in pod")
+				Expect(emulationFlagFound).To(BeTrue(), "Expected VM pod to have '--allow-emulation' flag")
+			})
+
+			It("should be reflected in domain XML", func() {
+				err := virtClient.RestClient().Post().Resource("virtualmachines").Namespace(tests.NamespaceTestDefault).Body(vm).Do().Error()
+				Expect(err).To(BeNil())
+
+				listOptions := metav1.ListOptions{}
+
+				Eventually(func() int {
+					podList, err := virtClient.CoreV1().Pods(tests.NamespaceTestDefault).List(listOptions)
+					Expect(err).ToNot(HaveOccurred())
+					return len(podList.Items)
+				}, 75, 0.5).Should(Equal(1))
+
+				Eventually(func() error {
+					podList, err := virtClient.CoreV1().Pods(tests.NamespaceTestDefault).List(listOptions)
+					Expect(err).ToNot(HaveOccurred())
+					for _, item := range podList.Items {
+						if strings.HasPrefix(item.Name, vm.ObjectMeta.GenerateName) {
+							return nil
+						}
+					}
+					return fmt.Errorf("Associated pod for VM '%s' not found", vm.Name)
+				}, 75, 0.5).Should(Succeed())
+
+				getOptions := metav1.GetOptions{}
+				var newVm *v1.VirtualMachine
+
+				newVm, err = virtClient.VM(tests.NamespaceTestDefault).Get(vm.Name, getOptions)
+				Expect(err).ToNot(HaveOccurred())
+
+				domain := &api.Domain{}
+				context := &api.ConverterContext{
+					AllowEmulation: true,
+					VirtualMachine: newVm,
+				}
+				api.Convert_v1_VirtualMachine_To_api_Domain(newVm, domain, context)
+
+				expectedType := ""
+				if _, err := os.Stat("/dev/kvm"); os.IsNotExist(err) {
```

Comment:
So I would expect, that this this is never true in our test environment. There will always be a /dev/kvm device, right?

Reply:
I agree that it would be odd, but this logic shadows the rules used in the code. I feel like it would be better not to fail over something we can anticipate and plan for.